### PR TITLE
GODRIVER-1456 make extJSONValueWriter escape keys

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: bson/bsonrw/extjson_writer.go [hunk 1]
```diff
@@ -536,7 +536,10 @@ func (ejvw *extJSONValueWriter) WriteUndefined() error {
 func (ejvw *extJSONValueWriter) WriteDocumentElement(key string) (ValueWriter, error) {
 	switch ejvw.stack[ejvw.frame].mode {
 	case mDocument, mTopLevel, mCodeWithScope:
-		ejvw.buf = append(ejvw.buf, []byte(fmt.Sprintf(`"%s":`, key))...)
+		var buf bytes.Buffer
+		writeStringWithEscapes(key, &buf, ejvw.escapeHTML)
```
Comment (mattChiaravalloti) [inline bson/bsonrw/extjson_writer.go]: Will this also ensure that keys like `"\usb#"` will be escaped? I ran into a problem with that specifically and then when I saw this bug I came up with the simpler test case you made above.

I figure that should be written as `"\\usb#"`? Is it worth adding a test case to make sure this works as well?

File: bson/bson_test.go [hunk 2]
```diff
@@ -111,3 +111,17 @@ func TestD(t *testing.T) {
...
+	doc := D{{Key: "\"", Value: 1}}
+	b, err := MarshalExtJSON(&doc, false, false)
+	noerr(t, err)
+
+	want := "{\"\\\"\":1}"
+	if diff := cmp.Diff(want, string(b)); diff != "" {
+		t.Errorf("Marshaled documents do not match. got %v, want %v", string(b), want)
+	}
+
+	err = UnmarshalExtJSON(b, false, &doc)
```
Comment (jyemin) [inline bson/bson_test.go]: To also ensure that the driver is unescaping the keys correctly, unmarshal into a different document and assert that it's equals to `doc`.

File: bson/bsonrw/extjson_writer.go (hunk 1 above)
Comment (iwysiu) [reply bson/bsonrw/extjson_writer.go]: I just changed the test case to the more complex one.

File: bson/bson_test.go (hunk 2 above)
Comment (iwysiu) [reply bson/bson_test.go]: done

File: bson/bsonrw/extjson_writer.go (hunk 1 above)
Comment (mattChiaravalloti) [reply bson/bsonrw/extjson_writer.go]: cool!

**Pull request merged**